src/android/Geolocation.java

```diff
@@ -303,4 +318,39 @@ public void onFailure(@NonNull Exception e) {
         task.addOnSuccessListener(checkLocationSettingsOnSuccess);
         task.addOnFailureListener(checkLocationSettingsOnFailure);
     }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        super.onActivityResult(requestCode, resultCode, intent);
+
+        LocationContext lc = locationContexts.get(requestCode);
+
+        switch (resultCode) {
+            case Activity.RESULT_CANCELED: {
+                PluginResult result = new PluginResult(PluginResult.Status.ERROR, LocationError.LOCATION_ENABLE_REQUEST_DENIED.toJSON());
+                lc.getCallbackContext().sendPluginResult(result);
+                locationContexts.delete(lc.getId());
+                requestForResolvable = null;
+                break;
+            }
+            case Activity.RESULT_OK: {
+                // request location updates because location was enabled
+                if (requestForResolvable != null) {
+                    requestLocationUpdates(lc, requestForResolvable);
+                    requestForResolvable = null;
+                }
+            }
+        }
+    }
+
+    /**
+     * Gets an hashCode from a String, making it a positive value if necessary
+     * @param id String to be hashed
+     * @return hashCode as a positive integer value
+     */
+    private int getPositiveHashCode(String id) {
+        int idHashCode = id.hashCode();
+        return idHashCode < 0 ? - idHashCode : idHashCode;
+    }
```

Comment:
The solution is a bit error-prone. What happens if we have two `LocationContext`s , one with the `123456` hash code and another with `-123456`?

Reply:
Yup, although it's probably unlikely, that can happen. I refactored the code to avoid doing this, can you check?